src/prefect/client/utilities.py

```diff
@@ -71,9 +71,23 @@ async def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
     return wrapper
 
 
+@overload
 def inject_client(
     fn: Callable[P, Coroutine[Any, Any, R]],
 ) -> Callable[P, Coroutine[Any, Any, R]]:
+    ...
+
+
+@overload
+def inject_client(
+    fn: Callable[P, R],
+) -> Callable[P, R]:
+    ...
+
+
+def inject_client(
+    fn: Callable[P, Union[Coroutine[Any, Any, R], R]],
+) -> Callable[P, Union[Coroutine[Any, Any, R], R]]:
```

Comment:
@desertaxle Why did you add this overload?

The implementation uses `await` on `fn()` so it **must** produce an awaitable, and nowhere in the prefect codebase is `@inject_client` used on a regular function.

Reply:
Yep, you're absolutely right. I was working with `@inject_client` while I was making `Block.load` sync. This was from something that ultimately didn't work, and I forgot to remove it. I'll make a note to remove this, but feel free to remove it if you're in the neighborhood.

Reply:
@desertaxle already on it 😁  See #16463.